ixmp/utils.py

```diff
@@ -34,6 +34,13 @@ def islistable(x):
     return isinstance(x, collections.Iterable) and not isstr(x)
 
 
+def isyear(y, s):
```

Comment:
This function seems clunky and doesn't provide the same functionality/interface as the rest of our `is*()` functions which simply say `True` or `False`. I think we would save more in code understanding if the two blocks that use this simply expand to:

```
if y is not None and isinstanc(y, int):
    do_stuff()
elif y is not None:
    raise error about type
```

Reply:
And just remove this line - I think in total we're saving like 2-4 lines by using this function at the moment. If we notice it gets used much more frequently, then we can reassess.

Reply:
I see now it's also used one time in the iiasa/message_ix#205, but my preference here remains.

Reply:
changed to `check_year()` as discussed offline